### PR TITLE
Adding check process to alignment at apply pars time

### PR DESCRIPTION
**Pull request by briochh**
Adding additional data store during add pars and checking against as apply pars time. 

Hopefully should catch misalignment of`index_cols` when applying pars.

Caveat: might cause unexpected failures... may need refining.

### CODE REVIEW

File: pyemu/utils/pst_from.py [hunk 1]
```diff
@@ -968,7 +968,7 @@ def _par_prep(
                         sep = " "
                         if rel_filepath.suffix.lower() == ".csv":
                             sep = ","
-                if df.columns.is_integer():
+                if pd.api.types.is_integer_dtype(df.columns):  # df.columns.is_integer(): # really!???
```
Comment (jtwhite79) [inline pyemu/utils/pst_from.py:971]: yikes!

**Review by jtwhite79 — APPROVED** [review]
Looks good to me.  I think having this check will help identify bugs sooner...

File: pyemu/utils/pst_from.py (hunk 1 above)
Comment (briochh) [reply pyemu/utils/pst_from.py:971]: vile ey?

Comment (briochh) [reply pyemu/utils/pst_from.py:971]: https://github.com/pandas-dev/pandas/issues/50042

**Pull request merged**